nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/WriteableView.java

```diff
@@ -736,11 +762,31 @@ public synchronized boolean remove(Object object) {
         return removed;
     }
 
-    public Object set(int index, Object object) {
-        Object replaced = proxied.set(index, object);
+    @Override
+    public E remove(int index) {
+        if (index >= size())
+                throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size());
+        E removed = proxied.get(index);
+        return remove(removed) ? removed : null;
+    }
+
+    @Override
+    public boolean removeIf(Predicate<? super E> filter) {
+        boolean removedAny = false;
+        for (E e : proxied) {
+            if (filter.test(e) && remove(e)) {
+                removedAny = true;
+            }
+        }
+        return removedAny;
+    }
+
+    @Override
+    public E set(int index, E object) {
```

Comment:
NB. While the above two now work I think `set` is not behaving properly. I looked into that but decided to not mess with it and avoid using it.